pkg/node/address.go

```diff
@@ -688,6 +692,26 @@ func GetWireguardPubKey() string {
 	return wireguardPubKey
 }
 
+// SetEndpointHealthIPv4 sets the IPv6 cilium-health endpoint address.
+func SetEndpointHealthIPv4(ip net.IP) {
+	ipv4HealthIP = ip
+}
+
+// GetEndpointHealthIPv4 returns the IPv4 cilium-health endpoint address.
+func GetEndpointHealthIPv4() net.IP {
+	return ipv4HealthIP
+}
+
+// SetEndpointHealthIPv6 sets the IPv6 cilium-health endpoint address.
+func SetEndpointHealthIPv6(ip net.IP) {
+	ipv6HealthIP = ip
+}
+
+// GetEndpointHealthIPv6 returns the IPv6 cilium-health endpoint address.
+func GetEndpointHealthIPv6() net.IP {
+	return ipv6HealthIP
+}
```

Comment:
I know that this style is more consistent with the way that most of `pkg/node` works, but I was hoping not to expand on this model. We used this style quite a bit across the Cilium codebase, and it's often one of the reasons that unit testing is difficult, because you can't just declare an interface & mock/dependency-inject such code.

Looking at the caller, previously the dependency here was explicitly passed into the initialization function, but now it calls this getter and the ordering is less explicit around how this value gets populated.  I think it's harder to trace both (1) initialization + (2) updates to see we are correctly handling the updates, especially now that this is more dynamic.

Have you thought about this too? I don't think this is appropriate to have in the discussion around this particular PR, so I don't want to block this. I would like to start a thread of discussion on some code maintenance / code debt items that I see like this though. At least hopefully get on the same page on whether we think it continues to make sense to accept code in this style or not.

Reply:
Yes, this also came up during the review session with Chris, Michi and Tom. I fully agree that we should get rid of the `pkg/node` pattern. Besides the problems you mentioned, we also we have two sources of truth for the local node (`pkg/node` and `pkg/nodediscovery`) which I think is a potential footgun.

My thinking here is also that we should get rid of the `pkg/node` getters. But I didn't want to tackle that refactor in this PR (both due to time constraints and the size of the PR). The sole reason I added getters to `pkg/node` instead of `pkg/nodediscovery.NodeDiscovery` is that at its more consistent with the existing code.

Let's create an issue or thread discussing how to address this technical debt in a follow-up PR.